gpflow/posteriors.py

```diff
@@ -70,27 +71,48 @@ def __init__(self, q_mu, q_sqrt):
         self.q_sqrt = q_sqrt  # [L, M, M], lower-triangular
 
 
+class PrecomputeCacheType(enum.Enum):
+    TENSOR = "tensor"
+    VARIABLE = "variable"
+    NOCACHE = "nocache"
+
+
+def _validate_precompute_cache_type(value) -> PrecomputeCacheType:
+    if isinstance(value, PrecomputeCacheType):
+        return value
+    elif isinstance(value, str):
+        return PrecomputeCacheType(value.lower())
+    elif not value:
+        return PrecomputeCacheType.NOCACHE
+    else:
+        raise ValueError(
+            f"{value} is not a valid PrecomputeCacheType. Valid options: 'tensor', 'variable', 'nocache' (or False)."
+        )
+
+
 class AbstractPosterior(Module, ABC):
     def __init__(
         self,
         kernel,
         inducing_variable,
-        q_mu,
-        q_sqrt,
-        whiten=True,
-        mean_function=None,
-        precompute=True,
+        q_mu: tf.Tensor,
+        q_sqrt: tf.Tensor,
+        whiten: bool = True,
+        mean_function: Optional[mean_functions.MeanFunction] = None,
+        *,
+        precompute_cache: PrecomputeCacheType,
     ):
+        """
+        Concrete implementations should not overwrite this constructor.
+
+        :param kernel:
```

Comment:
Is the intention to fill out this docstring?

Reply:
yeah ...